observatory.go

```diff
@@ -0,0 +1,403 @@
+	PerformanceScore int    `json:"performanceScore"`
+	State            string `json:"state"`
+	DeviceType       string `json:"deviceType"`
+	// TTFB is time to first byte
+	TTFB int `json:"ttfb"`
+	// FCP is first contentful paint
+	FCP int `json:"fcp"`
+	// LCP is largest contentful pain
+	LCP int `json:"lcp"`
+	// TTI is time to interactive
+	TTI int `json:"tti"`
+	// TBT is total blocking time
+	TBT int `json:"tbt"`
+	// SI is speed index
+	SI int `json:"si"`
+	// CLS is cumulative layout shift
+	CLS   float64          `json:"cls"`
+	Error *lighthouseError `json:"error,omitempty"`
+}
+
+// lighthouseError describes the test error.
+type lighthouseError struct {
+	Code              string `json:"code"`
+	Detail            string `json:"detail"`
+	FinalDisplayedURL string `json:"finalDisplayedUrl"`
+}
+
+// ObservatoryPageTrend describes the web vital metrics trend.
+type ObservatoryPageTrend struct {
+	PerformanceScore []*int     `json:"performanceScore"`
+	TTFB             []*int     `json:"ttfb"`
+	FCP              []*int     `json:"fcp"`
+	LCP              []*int     `json:"lcp"`
+	TTI              []*int     `json:"tti"`
+	TBT              []*int     `json:"tbt"`
+	SI               []*int     `json:"si"`
+	CLS              []*float64 `json:"cls"`
+}
+
+type ListObservatoryPagesParams struct {
+}
+
+// ObservatoryPagesResponse is the API response, containing a list of ObservatoryPage.
+type ObservatoryPagesResponse struct {
+	Response
+	Result []ObservatoryPage `json:"result"`
+}
+
+// ListObservatoryPages returns a list of pages which have been tested.
+//
+// API reference: https://api.cloudflare.com/#speed-list-pages
+func (api *API) ListObservatoryPages(ctx context.Context, rc *ResourceContainer, params ListObservatoryPagesParams) ([]ObservatoryPage, error) {
+	uri := fmt.Sprintf("/zones/%s/speed_api/pages", rc.Identifier)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return nil, err
+	}
+	var r ObservatoryPagesResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+	return r.Result, nil
+}
+
+type GetObservatoryPageTrendParams struct {
+	URL        string
+	Region     string     `url:"region"`
+	DeviceType string     `url:"deviceType"`
+	Start      time.Time  `url:"start"`
+	End        *time.Time `url:"end,omitempty"`
+	Timezone   string     `url:"tz"`
+	Metrics    []string
+}
+
+type ObservatoryPageTrendResponse struct {
+	Response
+	Result ObservatoryPageTrend `json:"result"`
+}
+
+// GetObservatoryPageTrend returns a the trend of web vital metrics for a page in a specific region.
+//
+// API reference: https://api.cloudflare.com/#speed-list-page-trend
+func (api *API) GetObservatoryPageTrend(ctx context.Context, rc *ResourceContainer, params GetObservatoryPageTrendParams) (*ObservatoryPageTrend, error) {
+	if params.URL == "" {
+		return nil, ErrMissingObservatoryUrl
+	}
+	v := url.Values{}
```

Comment:
i think majority of this can be removed if you use the [`buildURI`](https://github.com/cloudflare/cloudflare-go/blob/f968533e2861e3d84c52e6e5f194c31038604580/utils.go#L13) helper and `url` struct tags (example: https://github.com/cloudflare/cloudflare-go/blob/f968533e2861e3d84c52e6e5f194c31038604580/zone_hold.go#L30)